src/components/crossDevice/CrossDeviceLink/index.js

```diff
@@ -59,10 +63,11 @@ class CrossDeviceLink extends Component {
   }
 
   onGetConfig = (data) => {
-    const { roomId, mobileConfig, socket, nextStep } = this.props
+    const { roomId, mobileConfig, socket,actions, nextStep } = this.props
     if (roomId && roomId !== data.roomId) {
       socket.emit('leave', {roomId})
     }
+    actions.deleteMobileNumber()
```

Comment:
I don't like how we are using the presence of Mobile Number to determine what screen to show.

1) If I press back the app forgets my number, which is annoying
2) You end up with slightly hacky/illogical actions - why does receiving config mean I have to delete a mobile number from the store!

Reply:
Yeah, maybe it's a bit hack to display the message delivered component based on the mobile number presence... I will think of a way to refactor it.

Reply:
I think the main issue here is that our navigation is linear and I think we should fix that otherwise we will run into this type of issues more and more.